### PR TITLE
feature(follow and unfollow):user can follow and unfollow each other

### PR DESCRIPTION
**Pull request by EmyRukundo**
**What does this PR do?**

follow and unfollow each other

**Description of Task to be completed?**

user can follow and unfollow each other, but the user must exist and
user can't follow himself
use case doc link: https://docs.google.com/document/d/1wp7f0JTRJtrkgxYrGcop_POHw4SALw1Q2nOo5VjDsRA/edit

**How should this be manually tested?**

yarn run test

**Any background context you want to provide?**

N/A

**What are the relevant pivotal tracker stories?**

https://www.pivotaltracker.com/story/show/166241002

**Screenshots (if appropriate)**

**Questions:**

### CODE REVIEW

**Review by eliemugenzi — APPROVED** [review]

**Review by deschantkn — COMMENTED** [review]
Why did you change the `config/config.js` file ?
You also deleted ` src/sequelize/migrations/20190620145553-add-readtime.js` migration file ?

File: .env.example [hunk 1]
```diff
@@ -2,26 +2,26 @@ SECRET_KEY=
...
-DEV_DATABASE_URL=
-TEST_DATABASE_URL=
+DATABASE_URL_TEST=
 AUTHOSHAVEN_USER=
 AUTHOSHAVEN_PASS=
 BASE_URL=
 
-FCBK_ID=
-FCBK_APP_SECRET=
+FCBK_ID=392129074733761
```
Comment (deschantkn) [inline .env.example]: We don't put secret key values in `.env.example`

**Review by deschantkn — APPROVED** [review]

File: .env.example (hunk 1 above)
Comment (EmyRukundo) [reply .env.example]: Oooh my bad, corrected it

**Review by UhiriweAudace — APPROVED** [review]

File: .env.example [hunk 2]
```diff
@@ -2,27 +2,7 @@ SECRET_KEY=
 NODE_ENV=
 PORT=
 DATABASE_URL=
-DEV_DATABASE_URL=
-TEST_DATABASE_URL=
+DATABASE_URL_TEST=
```
Comment (Quantum-35) [inline .env.example]: Rename DB URL above to `TEST_DATABASE_URL`

File: src/api/controllers/profiles.js [hunk 3]
```diff
@@ -65,5 +74,128 @@ export default class ProfilesController {
...
+   * @Author - Audace Uhiriwe
+   * @param  {object} req
+   * @param  {object} res
+   * @return {object} returns an object containing the user's profiles
+   */
+  static async follow(req, res) {
+    const { username } = req.params;
+    const registeredUser = await User.findOne({ where: { username } });
+    if (registeredUser.id === req.user.id) {
+      return res.status(400).json({ errors: 'stop following yourself ' });
```
Comment (Quantum-35) [inline src/api/controllers/profiles.js]: Update the error message to `You can not follow yourself`

File: src/api/controllers/profiles.js [hunk 4]
```diff
@@ -65,5 +74,128 @@ export default class ProfilesController {
...
+          {
+            model: User,
+            as: 'follower',
+            attributes: ['id', 'firstName', 'lastName', 'email', 'username']
+          }
+        ]
+      })
+      .then((data) => {
+        if (!data[0]) {
+          return res.status(404).json({ error: 'no followers,you have now' });
```
Comment (Quantum-35) [inline src/api/controllers/profiles.js]: - Update the message to `you have no followers at the moment.`
- The status code should be 200.

File: src/api/controllers/profiles.js [hunk 5]
```diff
@@ -65,5 +74,128 @@ export default class ProfilesController {
...
+          {
+            model: User,
+            as: 'followedUser',
+            attributes: ['id', 'firstName', 'lastName', 'email', 'username']
+          }
+        ]
+      })
+      .then((data) => {
+        if (!data[0]) {
+          return res.status(404).json({ error: 'you do not follow anyone' });
```
Comment (Quantum-35) [inline src/api/controllers/profiles.js]: Update the Status code.

File: src/config/config.js [hunk 6]
```diff
@@ -3,23 +3,30 @@ import dotenv from 'dotenv';
...
+  dialect: 'postgres'
 };
 
 config.test = {
-  use_env_variable: 'TEST_DATABASE_URL',
-  logging: false,
+  username: process.env.DB_USERNAME,
+  password: process.env.DB_PASSWORD,
+  database: process.env.TEST_DB_NAME,
+  host: process.env.DB_HOST,
```
Comment (Quantum-35) [inline src/config/config.js]: Kindly use the Database url.

File: src/middleware/checkUsernameExist.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import models from '../sequelize/models';
+
+const usernameAvailability = {
+  async usernameExist(req, res, next) {
+    const { username } = req.params;
+    // console.log(username);
```
Comment (Quantum-35) [inline src/middleware/checkUsernameExist.js]: Work with @mifeille concerning this Middleware.

File: src/sequelize/migrations/20190620145553-add-readtime.js [hunk 8]
```diff
@@ -1,10 +0,0 @@
-
```
Comment (Quantum-35) [inline src/sequelize/migrations/20190620145553-add-readtime.js:1]: Pair with @eliemugenzi to ensure `Add read time` feature works without any glitches.

**Review by Quantum-35 — CHANGES_REQUESTED** [review]

File: src/middleware/shareArticle.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import open from 'open';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const { APP_FRONTEND_URL } = process.env.APP_FRONTEND_URL;
+const twitterShareURL = `https://twitter.com/intent/tweet?text=${APP_FRONTEND_URL}/articles/`;
```
Comment (eliemugenzi) [inline src/middleware/shareArticle.js]: This is your new feature, right? Why are you pushing it in this PR?

**Review by eliemugenzi — CHANGES_REQUESTED** [review]

File: src/middleware/shareArticle.js (hunk 9 above)
Comment (EmyRukundo) [reply src/middleware/shareArticle.js]: yes, was already started working on it but removed it, thanks

**Review by eliemugenzi — APPROVED** [review]

**Review by Quantum-35 — APPROVED** [review]

**Pull request merged**